lib/tests/vector.rs

```diff
@@ -14,7 +14,7 @@ async fn select_where_mtree_knn() -> Result<(), Error> {
 		CREATE pts:3 SET point = [8,9,10,11];
 		DEFINE INDEX mt_pts ON pts FIELDS point MTREE DIMENSION 4;
 		LET $pt = [2,3,4,5];
-		SELECT id, vector::distance::euclidean(point, $pt) AS dist FROM pts WHERE point <2> $pt;
+		SELECT id, vector::distance::euclidean(point, $pt) AS dist FROM pts WHERE point <2,EUCLIDEAN> $pt;
```

Comment:
This looks like we are forced to specify "euclidean" twice, one in "vector::distance::euclidean" and another in `<2,EUCLIDEAN>`. Why is that?

Can we imply `EUCLIDEAN` if `vector::distance::euclidean` is used, for instance?

Reply:
That could be a shortcut indeed, something close to what we did with "MatchRef" for the search functions.
But technically, Knn is treated differently, It does not rely on an index and it is more likely that users won't project the distance. I put it in the example to showcase the correctness of the operation.
Also, checking this match would add an overhead per query. It has been beneficial to the search functions due to the execution cost, but it would not be beneficial to Knn.